aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -145,12 +145,14 @@ module aptos_framework::code {
         let allowed_deps = check_dependencies(addr, &pack);
 
         // Check package against conflicts
+        // ....
         let module_names = get_module_names(&pack);
-        let packages = &mut borrow_global_mut<PackageRegistry>(addr).packages;
-        let len = vector::length(packages);
+        let package_immutable = &borrow_global<PackageRegistry>(addr).packages;
```

Comment:
@UIZorrot , is this code change necessary? or, can we avoid?

Reply:
this change is suggest by teng, which could Bypass the compiler error

Reply:
Yes, this is due to a compiler bug: https://github.com/aptos-labs/aptos-core/issues/8752

Reply:
@movekevin @xindingw We need to make the ref immutable before passing it to `enumerate_ref` below to make the prover work. Could you give some feedbacks here? Thanks

Reply:
It makes sense. Can you add a comment to clarify the purpose?

Reply:
@UIZorrot Could you add a comment to explain the purpose for this code change?

Reply:
surely